### PR TITLE
[cherry-pick]  Fix softmax cross entropy integer overflow.

### PR DESCRIPTION
**Pull request by ZHUI**
### PR types

Bug fixes
### PR changes

OPs
### Describe

Cherry-Pick #30590

### CODE REVIEW

**Review by wawltor — APPROVED** [review]
LGTM
